test/custom-field-validation-test.html

```diff
@@ -0,0 +1,114 @@
+<!doctype html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>vaadin-custom-field tests</title>
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <link rel="import" href="../../test-fixture/test-fixture.html">
+  <link rel="import" href="../vaadin-custom-field.html">
+</head>
+
+<body>
+  <test-fixture id="default">
+    <template>
+      <vaadin-custom-field>
+        <input type="text">
+        <input type="number">
+      </vaadin-custom-field>
+    </template>
+  </test-fixture>
+
+  <script>
+    describe('validation', function() {
+      var customField, inputElement;
+
+      function dispatchSlotChange() {
+        // NOTE(yuriy): Workaround not to use timeouts
+        const evt = new CustomEvent('slotchange');
+        customField.shadowRoot.querySelector('slot').dispatchEvent(evt);
+      }
+
+      beforeEach(function() {
+        customField = fixture('default');
+        inputElement = document.createElement('input');
+        dispatchSlotChange();
+      });
+
+      it('should propagate required property to internal inputs', function() {
+        expect(customField.__inputs.every(input => input.required)).to.be.false;
```

Comment:
it would be better to avoid testing internal API and use `querySelectorAll` in test instead.

Reply:
Changed.